pkg/scaler/client/client.go

```diff
@@ -154,6 +156,26 @@ func (c *WrapperClient) computers(ctx context.Context) (*gojenkins.Computers, er
 			return nil, fmt.Errorf("api response status code %d, body dump: %q", res.StatusCode, body)
 		}
 
+		if labels := c.opt.ExcludeNodesByLabel; len(labels) > 0 {
+			nodes := make([]*gojenkins.NodeResponse, 0)
+		OUTER:
+			for _, node := range computers.Computers {
+				for _, v := range node.AssignedLabels {
+					for _, label := range labels {
+						if v, ok := v["name"]; ok && v == label {
+							continue OUTER
+						}
+					}
+				}
+
+				nodes = append(nodes, node)
+			}
+
+			if len(nodes) > 0 {
+				computers.Computers = nodes
+			}
+		}
+
 		return computers, nil
```

Comment:
this is code is problematic, if `nodes` length is 0, it doesn't necessarily mean that it doesn't have agents at all, it can mean that only nodes with excluded labels are there, so you might return agents with excluded labels